### PR TITLE
Optimized GetClosestLimb functions

### PR DESCRIPTION
**Pull request by ZealanL**
Optimized GetClosestLimbOnRagdoll and GetClosestLimbOnSpritesheet, a previous TODO. Optimized version is much faster.

### CODE REVIEW

File: Barotrauma/BarotraumaClient/Source/Screens/CharacterEditorScreen.cs [hunk 1]
```diff
@@ -2377,20 +2377,40 @@ private void ValidateJoint(LimbJoint limbJoint)
...
-                .Where(l => filter == null ? true : filter(l))
-                .OrderBy(l => Vector2.Distance(SimToScreen(l.SimPosition), targetPos))
-                .FirstOrDefault();
+            Limb closestLimb = null;
+            float closestDistance = float.MaxValue;
+            foreach (Limb l in character.AnimController.Limbs) 
+            {
+                if (filter == null ? true : filter(l)) 
+                {
+                    float distance = Vector2.Distance(SimToScreen(l.SimPosition), targetPos);
```
Comment (juanjp600) [inline Barotrauma/BarotraumaClient/Source/Screens/CharacterEditorScreen.cs]: Could use DistanceSquared for a tiny extra improvement

Comment (thegrb93) [reply Barotrauma/BarotraumaClient/Source/Screens/CharacterEditorScreen.cs]: big extra improvement

Comment (ZealanL) [reply Barotrauma/BarotraumaClient/Source/Screens/CharacterEditorScreen.cs]: Yeah, that's a fair point - I was trying to retain the original functions used though.

Comment (itchyOwl) [reply Barotrauma/BarotraumaClient/Source/Screens/CharacterEditorScreen.cs]: gj

**Pull request merged**